src/components/permissions/PermissionIcon.js

```diff
@@ -0,0 +1,41 @@
+// @flow
+
+import React, { PropTypes } from 'react';
+import PermissionConsts from '../../core/permissions/Permission';
+import FontAwesome from 'react-fontawesome';
+
+export class PermissionIcon extends React.Component {
+  static propTypes = {
+    permission: PropTypes.instanceOf(PermissionConsts.Permission).isRequired,
+    size: PropTypes.oneOf(['lg', '2x', '3x', '4x', '5x'])
+  };
+
+  render() {
+    let size = this.props.size || 'lg';
+    let icon;
+
+    switch (this.props.permission) {
+      case PermissionConsts.OWNER:
+        icon = "user-o";
+        break;
+      case PermissionConsts.WRITE:
+        icon = "pencil";
+        break;
+      case PermissionConsts.READ:
+        icon = "eye";
+        break;
+      case PermissionConsts.DISCOVER:
+        icon = "lock";
+        break;
+      default:
+        return "";
```

Comment:
will this create a DOM element? if I understand this correctly, if we hit the `default` case, the icon should not be rendered. I take this to also mean that there should be no DOM element.